packages/components/src/DataList/components/DataListLayoutInternal/DataListHeaderCheckbox.tsx

```diff
@@ -42,14 +51,12 @@ export function DataListHeaderCheckbox({ children }: DataListHeaderCheckbox) {
         initialChild={children}
         switchTo={
           <div className={styles.headerBatchSelect}>
-            <Text>
-              <b>{selected.length} selected</b>
-            </Text>
```

Comment:
I'd actually keep this here so it's a part of the transition when it shows up. I'd override the `b` tags `font-weight` tho

Reply:
As for how VO would announce the selected, you can add a child with the same text value but style it to [only show for VO](https://webaim.org/techniques/css/invisiblecontent/#techniques:~:text=.sr%2Donly%20%7B%0Aposition%3Aabsolute%3B%0Aleft%3A%2D10000px%3B%0Atop%3Aauto%3B%0Awidth%3A1px%3B%0Aheight%3A1px%3B%0Aoverflow%3Ahidden%3B%0A%7D)

Reply:
Done!